internal/storage/storage_handle.go

```diff
@@ -77,11 +78,20 @@ func NewStorageHandle(ctx context.Context, clientConfig StorageClientConfig) (sh
 	}
 
 	// Custom http client for Go Client.
-	httpClient := &http.Client{
-		Transport: &oauth2.Transport{
-			Base:   transport,
-			Source: clientConfig.TokenSrc,
-		},
+	oauth2Transport := oauth2.Transport{
+		Base:   transport,
+		Source: clientConfig.TokenSrc,
+	}
+	var httpClient *http.Client
```

Comment:
As per the documentation, a time-out as 0 mean no timeout.

```Go
	// Timeout specifies a time limit for requests made by this
	// Client. The timeout includes connection time, any
	// redirects, and reading the response body. The timer remains
	// running after Get, Head, Post, or Do return and will
	// interrupt reading of the Response.Body.
	//
	// A Timeout of zero means no timeout.
	//
	// The Client cancels requests to the underlying Transport
	// as if the Request's Context ended.
	//
	// For compatibility, the Client will also use the deprecated
	// CancelRequest method on Transport if found. New
	// RoundTripper implementations should use the Request's Context
	// for cancellation instead of implementing CancelRequest.
	Timeout time.Duration
```

Can you please check if it works, in that case we don't need special check?

Reply:
Yes, that worked. Made changes accordingly.